quickwit/quickwit-proto/protos/quickwit/ingester.proto

```diff
@@ -24,24 +24,31 @@ package quickwit.ingest.ingester;
 import "quickwit/ingest.proto";
 
 service IngesterService {
-  // Persists batches of documents to primary shards owned by a leader.
+  // Persists batches of documents to primary shards hosted on a leader.
   rpc Persist(PersistRequest) returns (PersistResponse);
 
   // Opens a replication stream from a leader to a follower.
   rpc OpenReplicationStream(stream SynReplicationMessage) returns (stream AckReplicationMessage);
 
-  // Streams records from a leader or a follower. The client can optionally specify a range of positions to fetch.
+  // Streams records from a leader or a follower. The client can optionally specify a range of positions to fetch,
+  // otherwise the stream will go undefinitely or until the shard is closed.
   rpc OpenFetchStream(OpenFetchStreamRequest) returns (stream FetchResponseV2);
 
-  // rpc OpenWatchStream(OpenWatchStreamRequest) returns (stream WatchMessage);
+  // Streams status updates, called "observations", from an ingester.
+  rpc OpenObservationStream(OpenObservationStreamRequest) returns (stream ObservationMessage);
 
+  // Closes a set of shards.
   rpc CloseShards(CloseShardsRequest) returns (CloseShardsResponse);
 
   // Pings an ingester to check if it is ready to host shards and serve requests.
   rpc Ping(PingRequest) returns (PingResponse);
 
-  // Truncates the shards at the given positions. Indexers should call this RPC on leaders, which will replicate the request to followers.
+  // Truncates the shards at the given positions.
```

Comment:
With replication is the information conveyed somehow to the replication peer?
(either way it is worth commenting here - or somewhere else- I think)

Reply:
No, indexers must call truncate directly on leaders and followers. I'll update the doc.